### ISSUE TITLE
exception: WebSocket is not open: readyState 3 (CLOSED)

### ISSUE THREAD

**Issue by zixia** [lijiarui/wechaty-puppet-padchat#164]
Did our server ok now?

```
18:25:09 VERB PadchatRpc puppet-padchat is disconnected, rpcCall(WXGetContact) is scheduled. This call will be made again when connection restored
18:25:09 SILL PuppetPadchat contactPayload(wxid_a3iaf6nsyhd822) exception: WebSocket is not open: readyState 3 (CLOSED)
18:25:09 SILL PuppetPadchat contactPayload(wxid_a3iaf6nsyhd822) get failed for undefined, try load from room member data source
18:25:09 SILL PuppetPadchat contactRawPayload(wxid_a3iaf6nsyhd822)
18:25:09 SILL PuppetPadchatManager contactRawPayload(wxid_a3iaf6nsyhd822)
18:25:09 SILL PuppetPadchatManager contactRawPayload(wxid_a3iaf6nsyhd822) retry() attempt=1
18:25:09 SILL PadchatRpc rpcCall(WXGetContact, ["wxid_a3iaf6nsyhd822"])
18:25:09 VERB PadchatRpc puppet-padchat is disconnected, rpcCall(WXGetContact) is scheduled. This call will be made again when connection restored
18:25:09 SILL PuppetPadchat contactPayload(wxid_gimogrsh5djr21) exception: WebSocket is not open: readyState 3 (CLOSED)
18:25:09 SILL PuppetPadchat contactPayload(wxid_gimogrsh5djr21) get failed for undefined, try load from room member data source
18:25:09 SILL PuppetPadchat contactRawPayload(wxid_gimogrsh5djr21)
18:25:09 SILL PuppetPadchatManager contactRawPayload(wxid_gimogrsh5djr21)
18:25:09 SILL PuppetPadchatManager contactRawPayload(wxid_gimogrsh5djr21) retry() attempt=1
18:25:09 SILL PadchatRpc rpcCall(WXGetContact, ["wxid_gimogrsh5djr21"])
18:25:09 VERB PadchatRpc puppet-padchat is disconnected, rpcCall(WXGetContact) is scheduled. This call will be made again when connection restored
18:25:09 SILL PuppetPadchat contactPayload(wxid_cvy37oplqqrl22) exception: WebSocket is not open: readyState 3 (CLOSED)
18:25:09 SILL PuppetPadchat contactPayload(wxid_cvy37oplqqrl22) get failed for undefined, try load from room member data source
18:25:09 SILL PuppetPadchat contactRawPayload(wxid_cvy37oplqqrl22)
18:25:09 SILL PuppetPadchatManager contactRawPayload(wxid_cvy37oplqqrl22)
18:25:09 SILL PuppetPadchatManager contactRawPayload(wxid_cvy37oplqqrl22) retry() attempt=1
18:25:09 SILL PadchatRpc rpcCall(WXGetContact, ["wxid_cvy37oplqqrl22"])
18:25:09 VERB PadchatRpc puppet-padchat is disconnected, rpcCall(WXGetContact) is scheduled. This call will be made again when connection restored
(node:57354) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 4054)
18:26:02 ERR Config ###########################
18:26:02 ERR Config unhandledRejection: Error: WebSocket is not open: readyState 3 (CLOSED) [object Promise]
18:26:02 ERR Config ###########################
18:26:02 ERR Config process.on(unhandledRejection) promise.catch(WebSocket is not open: readyState 3 (CLOSED))
Config Error: WebSocket is not open: readyState 3 (CLOSED)
    at WebSocket.send (/home/ubuntu/workspace/node_modules/ws/lib/websocket.js:314:19)
    at Peer.jsonRpc.on (/home/ubuntu/workspace/node_modules/wechaty-puppet-padchat/src/padchat-rpc.ts:246:19)
    at Peer.emit (events.js:182:13)
    at Peer.EventEmitter.emit (domain.js:442:20)
    at Peer.emit (/home/ubuntu/workspace/node_modules/json-rpc-peer/src/index.js:197:14)
    at push (/home/ubuntu/workspace/node_modules/json-rpc-peer/src/index.js:142:12)
    at new Promise (<anonymous>)
    at Peer.request (/home/ubuntu/workspace/node_modules/json-rpc-peer/src/index.js:139:12)
    at Timeout.setTimeout (/home/ubuntu/workspace/node_modules/wechaty-puppet-padchat/src/padchat-rpc.ts:193:26)
    at ontimeout (timers.js:427:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
    at Timer.processTimers (timers.js:212:10)
(node:57354) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 4056)
18:27:03 SILL Puppet constructor() watchdog.on(reset) reason: {"data":"ws.on(message)","timeout":240000}
18:27:03 SILL Puppet constructor() this.on(reset) reason: {"data":"ws.on(message)","timeout":240000}
18:27:03 SILL Puppet constructor() resetThrottleQueue.subscribe() reason: {"data":"ws.on(message)","timeout":240000}
18:27:03 VERB Puppet reset({"data":"ws.on(message)","timeout":240000})
18:27:03 SILL StateSwitch <PuppetPadchat> off() is false
18:27:03 VERB PuppetPadchat stop()
18:27:03 SILL StateSwitch <PuppetPadchat> off() is false
18:27:03 SILL StateSwitch <PuppetPadchat> off() is false
18:27:03 VERB StateSwitch <PuppetPadchat> off(pending) <- (false)
18:27:03 VERB PuppetPadchat logout()
18:27:03 SILL Contact ready() @ Puppet#0<PuppetPadchat>(ding-dong-bot)/PuppetPadchat#0
18:27:03 SILL Contact ready() isReady() true
18:27:03 VERB PuppetPadchatManager logout()
18:27:03 VERB PuppetPadchatManager releaseCache()
18:27:03 SILL PuppetPadchatManager releaseCache() closing caches ...
18:27:03 VERB ContactSelf name()
李卓桓 logouted
```

**Comment by windmemory** [comment]
This issue should be fixed with commit: 03fa49fd008f92ec611b7363198579291c319876

Please verify if the problem still exits, if not, please help us close this issue :)

**Comment by zixia** [comment]
Awesome. Will close this issue for now, cheers!
